src/gui/application.cpp

```diff
@@ -551,51 +544,105 @@ namespace Toolbox {
+                if (maybe_filters) {
+                    auto filters = maybe_filters.value();
+                    num_filters  = filters.size();
+                    nfd_filters  = new nfdnfilteritem_t[num_filters];
+                    for (int i = 0; i < num_filters; ++i) {
+                        std::wstring filter_name =
+                            std::wstring(filters[i].first.begin(), filters[i].first.end());
+                        std::wstring filter_ext =
+                            std::wstring(filters[i].second.begin(), filters[i].second.end());
+                        m_filters.push_back({filter_name, filter_ext});
+                        nfd_filters[i] = {m_filters[i].first.c_str(), m_filters[i].second.c_str()};
+                    }
+                }
+                nfdopendialognargs_t args;
+                args.filterList  = const_cast<const nfdnfilteritem_t *>(nfd_filters);
+                args.filterCount = num_filters;
+                args.defaultPath = m_starting_path.c_str();
+                NFD_GetNativeWindowFromGLFWWindow(parent_window, &args.parentWindow);
+                m_result = NFD_OpenDialogN_With(&m_selected_path, &args);
+                if (maybe_filters) {
+                    delete[] nfd_filters;
+                }
+            }
+            m_thread_running = false;
+        };
+        m_thread = std::thread(fn);
+    }
+#else
+    void FileDialog::OpenDialog(
+        std::filesystem::path starting_path, GLFWwindow *parent_window, bool is_directory,
+        std::optional<std::vector<std::pair<std::string, std::string>>> maybe_filters) {
         if (m_thread_initialized) {
             m_thread.join();
         } else {
             m_thread_initialized = true;
         }
         m_thread_running = true;
-        auto fn = [this, starting_path, parent_window, is_directory, maybe_filters]
-                               () {
-            if(is_directory){
+        m_closed         = false;
+        auto fn          = [this, starting_path, parent_window, is_directory, maybe_filters]() {
+            m_starting_path = starting_path.wstring();
+            m_filters.clear();
+            if (is_directory) {
                 nfdpickfoldernargs_t args;
-                args.defaultPath = starting_path.string().c_str();
+                args.defaultPath = m_starting_path.c_str();
                 NFD_GetNativeWindowFromGLFWWindow(parent_window, &args.parentWindow);
                 m_result = NFD_PickFolderN_With(&m_selected_path, &args);
-                //m_result = NFD_PickFolderN(&m_selected_path, starting_path.string().c_str());
+                // m_result = NFD_PickFolderN(&m_selected_path, starting_path.string().c_str());
             } else {
-                int num_filters = 0;
-                nfdu8filteritem_t* nfd_filters = nullptr;
-                if(maybe_filters) {
+                int num_filters               = 0;
+                nfdnfilteritem_t *nfd_filters = nullptr;
```

Comment:
Ah yes, a classic C-ism to put the * on the right meaning "the thing pointed to by nfd_filters is a nfdfilter_item_t" instead of "the type of nfd_filters is pointer-to-nfdfilter_item_t". I'll try to follow that.

Reply:
Also on these lines: on the linux build of nativefiledialogs-extended nfdnfilteritem_t is an alias for nfdu8filteritem_t. Since we're already in a platform specific branch, is there a reason to change to the "n" version here? Just to indicate that u8's are the native char type on Linux?